docs/apis/core/dml/index.md

```diff
@@ -0,0 +1,1066 @@
+
+```php
+public function get_fieldset_select(
+    $table,
+    $return,
+    $select,
+    array $params = null
+);
+```
+
+### get_fieldset_sql
+
+Return values of the first column as an array using a custom SELECT field FROM ... query.
+
+```php
+public function get_fieldset_sql(
+    $sql,
+    array $params = null
+);
+```
+
+## Setting a field value
+
+### set_field
+
+Set a single field in every record where all the given conditions are met.
+
+```php
+public function set_field(
+    $table,
+    $newfield,
+    $newvalue,
+    array $conditions = null
+);
+```
+
+### set_field_select
+
+Set a single field in every table record where the given conditions are used in the WHERE clause.
+
+```php
+public function set_field_select(
+    $table,
+    $newfield,
+    $newvalue,
+    $select,
+    array $params = null
+);
+```
+
+## Deleting records
+
+### delete_records
+
+Delete records from the table where all the given conditions are met.
+
+```php
+public function delete_records(
+    $table,
+    array $conditions = null
+);
+```
+
+### delete_records_select
+
+Delete records from the table where the given conditions are used in the WHERE clause.
+
+```php
+public function delete_records_select(
+    $table,
+    $select,
+    array $params = null
+);
+```
+
+## Inserting records
+
+### insert_record
+
+Insert the given data object into the table and return the "id" of the newly created record.
+
+```php
+public function insert_record(
+    $table,
+    $dataobject,
+    $returnid = true,
+    $bulk = false
+);
+```
+
+### insert_records
+
+Insert multiple records into the table as fast as possible. Records are inserted in the given order, but the operation is not atomic. Use transactions if necessary.
+
+```php
+public function insert_records(
+    $table,
+    $dataobjects
+);
+```
+
+### insert_record_raw
+
+For rare cases when you also need to specify the ID of the record to be inserted.
+
+## Updating records
+
+### update_record
+
+Update a record in the table. The data object must have the property "id" set.
+
+```php
+public function update_record(
+    $table,
+    $dataobject,
+    $bulk = false
+);
+```
+
+## Executing a custom query
+
+### execute
+
+- If you need to perform a complex update using arbitrary SQL, you can use the low level "execute" method. Only use this when no specialised method exists.
+
+```php
+public function execute(
+    $sql,
+    array $params = null
+);
+```
+
+:::danger
+
+Do NOT use this to make changes in database structure, use the `database_manager` methods instead.
+
+:::
+
+## Using recordsets
+
+If the number of records to be retrieved from DB is high, the 'get_records_xxx() functions above are far from optimal, because they load all the records into the memory via the returned array. Under those circumstances, it is highly recommended to use these get_recordset_xxx() functions instead. They return an iterator to iterate over all the found records and save a lot of memory.
```

Comment:
```suggestion
If the number of records to be retrieved from DB is high, the `get_records_xxx()` functions above are far from optimal, because they load all the records into the memory via the returned array. Under those circumstances, it is highly recommended to use these `get_recordset_xxx()` functions instead. They return an iterator to iterate over all the found records and save a lot of memory.
```